packages/config/src/getConfig.ts

```diff
@@ -4,26 +4,37 @@ import { spawnSync } from 'child_process';
 import { AppJSONConfig, ConfigContext, ExpoConfig } from './Config.types';
 import { ConfigError, errorFromJSON } from './Errors';
 import { serializeAndEvaluate } from './Serialize';
-import { DynamicConfigResults, evalConfig } from './evalConfig';
+import { DynamicConfigResults } from './evalConfig';
 
 function isMissingFileCode(code: string): boolean {
   return ['ENOENT', 'MODULE_NOT_FOUND', 'ENOTDIR'].includes(code);
 }
 
+let _cachedConfig: DynamicConfigResults | null = null;
+
+// Force re-evaluation of dynamic config
+export function clearDynamicConfigCache() {
+  _cachedConfig = null;
+}
+
 // We cannot use async config resolution right now because Next.js doesn't support async configs.
 // If they don't add support for async Webpack configs then we may need to pull support for Next.js.
 function readConfigFile(configFile: string, context: ConfigContext): null | DynamicConfigResults {
   try {
     if (context.useDynamicEval) {
       return spawnAndEvalConfig(configFile, context);
+    } else if (!context.useDynamicEval && _cachedConfig === null) {
+      _cachedConfig = spawnAndEvalConfig(configFile, context);
```

Comment:
If someone wants to reuse a stale config they can always just limit how often they invoke getConfig. Moving that logic into getConfig seems like it’ll break stuff.